lib/ansible/modules/cloud/amazon/ec2_vol.py

```diff
@@ -221,150 +224,160 @@
     }
 '''
 
-import time
-
-from distutils.version import LooseVersion
-
 try:
-    import boto
-    import boto.ec2
-    import boto.exception
-    from boto.exception import BotoServerError
-    from boto.ec2.blockdevicemapping import BlockDeviceType, BlockDeviceMapping
+    import botocore
 except ImportError:
-    pass  # Taken care of by ec2.HAS_BOTO
+    pass  # Handled by AnsibleAWSModule
+
+from ansible.module_utils.aws.core import AnsibleAWSModule
+from ansible.module_utils.ec2 import (
+    AWSRetry,
+    ec2_argument_spec,
+    camel_dict_to_snake_dict,
+    boto3_tag_list_to_ansible_dict,
+    ansible_dict_to_boto3_filter_list,
+    ansible_dict_to_boto3_tag_list,
+    compare_aws_tags
+)
+
+
+def get_instance(module, ec2_conn, instance_id=None):
+    instance = None
+    if not instance_id:
+        return instance
+
+    try:
+        reservation_response = ec2_conn.describe_instances(InstanceIds=[instance_id])
+        instance = camel_dict_to_snake_dict(reservation_response['Reservations'][0]['Instances'][0])
+    except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+        module.fail_json_aws(e, msg='Error while getting Instance wiht id {0}'.format(instance))
 
-from ansible.module_utils.basic import AnsibleModule
-from ansible.module_utils.ec2 import (HAS_BOTO, AnsibleAWSError, connect_to_aws, ec2_argument_spec,
-                                      get_aws_connection_info)
+    return instance
 
 
-def get_volume(module, ec2):
+def get_volume(module, ec2_conn, vol_id=None, fail_on_not_found=True):
     name = module.params.get('name')
-    id = module.params.get('id')
+    param_id = module.params.get('id')
     zone = module.params.get('zone')
-    filters = {}
-    volume_ids = None
+
+    if not vol_id:
+        vol_id = param_id
 
     # If no name or id supplied, just try volume creation based on module parameters
-    if id is None and name is None:
+    if vol_id is None and name is None:
         return None
 
-    if zone:
-        filters['availability_zone'] = zone
-    if name:
-        filters = {'tag:Name': name}
-    if id:
-        volume_ids = [id]
+    find_params = dict()
+    vols = []
+    if vol_id:
+        find_params['VolumeIds'] = [vol_id]
+    elif name:
+            find_params['Filters'] = ansible_dict_to_boto3_filter_list({'tag:Name': name})
+    elif zone:
+            find_params['Filters'] = ansible_dict_to_boto3_filter_list({'availability-zone': zone})
+
     try:
-        vols = ec2.get_all_volumes(volume_ids=volume_ids, filters=filters)
-    except boto.exception.BotoServerError as e:
-        module.fail_json(msg="%s: %s" % (e.error_code, e.error_message))
+        vols_response = ec2_conn.describe_volumes(**find_params)
```

Comment:
You should probably use the paginator here in case they haven't provided volume IDs and there are too many responses. https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ec2.html#EC2.Paginator.DescribeVolumes